pymc_marketing/mlflow.py

```diff
@@ -412,14 +463,320 @@ def log_inference_data(
+    artifact_path : str, optional
+        The path to the artifact to be logged. Defaults to "mmm_model".
+    conda_env : dict, optional
+        A dictionary representation of a Conda environment. Defaults to the default conda environment.
+    registered_model_name : str, optional
+        The name of the registered model to be logged. Defaults to None.
+        If specified, the model will be registered under this name, otherwise it will not be registered.
+    wrapper_kwargs : dict, optional
+        Additional keyword arguments to pass to the MMMWrapper, controlling inference behaviour of logged model.
+
+    Examples
+    --------
+    MLFlow Registering for a PyMC-Marketing model:
+
+    .. code-block:: python
+
+        import pandas as pd
+
+        import mlflow
+
+        from pymc_marketing.mmm import (
+            GeometricAdstock,
+            LogisticSaturation,
+            MMM,
+        )
+        import pymc_marketing.mlflow
+        from pymc_marketing.mlflow import MMMWrapper
+
+        pymc_marketing.mlflow.autolog(log_mmm=True)
+
+        # Usual PyMC-Marketing model code
+
+        data_url = "https://raw.githubusercontent.com/pymc-labs/pymc-marketing/main/data/mmm_example.csv"
+        data = pd.read_csv(data_url, parse_dates=["date_week"])
+
+        X = data.drop("y",axis=1)
+        y = data["y"]
+
+        mmm = MMM(
+            adstock=GeometricAdstock(l_max=8),
+            saturation=LogisticSaturation(),
+            date_column="date_week",
+            channel_columns=["x1", "x2"],
+            control_columns=[
+                "event_1",
+                "event_2",
+                "t",
+            ],
+            yearly_seasonality=2,
+        )
+
+        mlflow.set_experiment("MMM Experiment")
+
+        with mlflow.start_run():
+            idata = mmm.fit(X, y)
+
+            # Additional specific logging
+            fig = mmm.plot_components_contributions()
+            mlflow.log_figure(fig, "components.png")
+
+            model_info = log_model(mmm,
+                    registered_model_name="my_amazing_mmm",
+                    include_last_observations=True,
+                    original_scale=False,
+                    )
+    """
+    # Incorporate MMM into MLflow workflow
+    mlflow_mmm = MMMWrapper(
+        mmm,
+        extend_idata=wrapper_kwargs.get("extend_idata", False),
+        combined=wrapper_kwargs.get("combined", True),
+        include_last_observations=wrapper_kwargs.get(
+            "include_last_observations", False
+        ),
+        original_scale=wrapper_kwargs.get("original_scale", True),
+    )
+
+    mlflow.pyfunc.log_model(
+        artifact_path=artifact_path,
+        python_model=mlflow_mmm,
+    )
+    run_id = mlflow.active_run().info.run_id
+    model_uri = f"runs:/{run_id}/{artifact_path}"
+
+    if registered_model_name:
+        mlflow.register_model(model_uri, registered_model_name)
+
+
+def load_model(
+    run_id: str | None = None,
```

Comment:
This shouldn't allow None unless we get the active run from the context. Do other mmlflow load functions get the run_id as well?